### PR TITLE
Plan: OxiSynth processor integration

### PR DESCRIPTION
**Pull request by SanderVocke**
### Motivation

- Define a clear, reviewable implementation plan to add a new built-in SoundFont synth processor, OxiSynth, usable on native and WebAssembly builds and controlled entirely via MIDI. 
- Capture immutable acceptance criteria, design constraints (including embedded `TimGM6mb.sf2`), and a staged rollout so engine, backend, worklet, session, and packaging work can proceed safely and reproducibly.

### Description

- Add a new plan file `plans/oxisynth_processor.md` that contains goals and scope, immutable acceptance criteria, design rules and constraints, the execution contract, and an ordered six-stage implementation checklist covering feasibility, engine integration, native and worklet backends, session persistence, integration, and end-to-end validation. 
- The plan specifies the processor identity (`oxisynth`), fixed topology (`0 dry audio / 2 wet audio / 1 dry MIDI`), transient synth state (no editor/state persistence for v1), sample-accurate MIDI sub-block rendering, embedding and provenance requirements for `TimGM6mb.sf2`, and a pinned `oxisynth` crate dependency with wasm compatibility as a requirement. 
- The plan requires asset integrity tests and no-allocation/no-realtime violations for the audio path, and enumerates verification steps and evidence required before declaring the feature complete.

### Testing

- Ran `git diff --check` to validate there are no whitespace/patch problems, which passed. 
- Performed repository sanity checks using `git status --short --branch` and `git log -1 --oneline` to confirm the new plan file is present and visible, which succeeded. 
- Exercised a local MCP `make_pr` invocation to prepare PR metadata (title/body) against the plan payload, which returned the expected PR metadata successfully. 
- No runtime or build tests were required for this plan-only change.

------
[Codex Task](https://chatgpt.com/codex/cloud/tasks/task_e_6a88662218a48328a37892e934a9a7f7)

### CODE REVIEW

File: plans/oxisynth_processor.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+- [ ] Add an OxiSynth backend variant to the engine processor route, lifecycle/activation hooks, port registration, processing dispatch, and teardown. Preserve ordered sample offsets by splitting rendering at event boundaries and write silence when inactive.
+- [ ] Add focused engine tests for stereo output, timing at block boundaries, multi-channel program/drum behavior, activation/reset/removal, malformed MIDI, sample-rate variation, bounded event capacity, and no allocation/no realtime lock violations.
+
+**Verification:** run the OxiSynth engine tests plus existing session scheduling, MIDI, no-allocation, and realtime-lock suites on the dummy backend.
+
+### Stage 3 — Shared catalog and native backend
+
+Depends on Stage 2.
+
+- [ ] Add `TrackProcessorTypeId::OXISYNTH` and an always-available descriptor with fixed constraints (`dry=0`, `wet=2`, required MIDI), no editor, and no persistent processor state/recovery/log features.
```
[the remark this answers is not in the document]
Comment (SanderVocke) [reply plans/oxisynth_processor.md]: Addressed in 88cd7f6: Stage 3 now explicitly extends the shared constraints model with minimum or exact channel bounds, updates all consumers while preserving existing behavior, and tests both undersized and oversized shapes.

File: plans/oxisynth_processor.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+- [ ] Instantiate and run the same engine wrapper inside `shoop_audio_worklet`, including the embedded SoundFont in the actual worklet wasm artifact; retain bounded quantum processing and transactional error reporting.
+- [ ] Extend protocol, client, worklet, and browser tests for creation, MIDI-driven stereo output, program changes, activation, removal/recreation, malformed input, rollback, and native/browser snapshot parity.
+
+**Verification:** run wasm unit/browser tests and dependency-isolation checks; build actual debug and release web artifacts and verify they contain no runtime SoundFont fetch/file access and remain within the repository's artifact-size budget (updating the documented budget deliberately for the embedded asset if required).
+
+### Stage 5 — Session persistence and generic application integration
+
+Depends on Stages 3 and 4.
+
+- [ ] Add an additive `OxiSynth` track topology and chain type to the version-1 session document, codec, validation, and backend conversion paths, representing only the fixed channel shape and processor identity.
```
[the remark this answers is not in the document]
Comment (SanderVocke) [reply plans/oxisynth_processor.md]: Addressed in 88cd7f6: Stage 5 now calls for session document version 4, current-version/archive-dispatch updates, and explicit migrations from supported versions 1 through 3.

File: plans/oxisynth_processor.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+- [ ] Audit or instrument OxiSynth construction, `send_event`, and `write` calls for realtime behavior; select fixed polyphony/buffer limits and record any required upstream workaround before graph integration.
+
+**Verification:** build the engine for native and `wasm32-unknown-unknown`; run the focused asset/render test; run license/metadata checks; prove the render path passes the repository's no-allocation guard after warm-up.
+
+### Stage 2 — Engine processor and realtime graph
+
+Depends on Stage 1.
+
+- [ ] Add an engine-owned OxiSynth processor wrapper that constructs from embedded bytes off the audio thread, validates sample-rate/channel settings, preallocates scratch storage, translates valid MIDI messages, and renders two output channels without allocation.
+- [ ] Specify and test MIDI translation for note on/off (including velocity-zero note-on), poly/channel pressure, CC/bank select, program change, pitch bend, all-notes-off/all-sound-off, and reset; safely reject truncated, SysEx, realtime, and unsupported system messages.
```
[the remark this answers is not in the document]
Comment (SanderVocke) [reply plans/oxisynth_processor.md]: Addressed in 88cd7f6: Stage 2 now explicitly handles MIDI System Reset (0xFF), rejects only the other unsupported realtime/system messages, and verifies reset releases active voices.

File: src/rust/shoop_worklet_client/src/lib.rs [hunk 4]
```diff
@@ -1071,6 +1077,46 @@ fn browser_tiny_port_descriptors(
...
+            id,
+            owner: BackendPortOwner::Track,
+            name: format!("{base}_audio_dry_in_{}", index + 1),
+            data_type: BackendPortDataType::Audio,
+            direction: BackendPortDirection::Input,
+            role: BackendPortRole::AudioInput,
+        });
+        let id = BackendPortId::from_raw(*next_port_id);
+        *next_port_id = next_port_id.saturating_add(1);
+        ports.push(BackendPortDescriptor {
```
[the remark this answers is not in the document]
Comment (SanderVocke) [reply src/rust/shoop_worklet_client/src/lib.rs:1100]: Addressed in 9524229: browser port reservations now group both dry inputs before both wet outputs, matching worklet registration order, with a regression test locking IDs, roles, and directions.

File: docs/session_format_v1.md [hunk 5]
```diff
@@ -75,6 +76,8 @@ Float WAV is the baseline standard cross-target audio format. The current native
 
 `TinySynthFx` stores one `audio_channels` count. It always means the same number of dry inputs and wet outputs plus exactly one dry MIDI input; zero audio channels is valid. Its chain type is also `TinySynthFx`. The stable runtime processor ID is `tiny_synth_fx`, while **Tiny Synth/FX** is only its display label.
 
+`OxiSynth` stores no variable channel fields. It always means zero dry audio inputs, exactly two wet audio outputs, and one dry MIDI input; its chain type is also `OxiSynth` and its stable runtime processor ID is `oxisynth`. The chain's `internal_state` must be empty, it may not contain Tiny Synth/FX MIDI mappings, and no automatic recorded-take `fx_state` is written. Document version 4 adds this topology; versions 1 through 3 remain accepted and migrate without changing their topology or processor state.
```
[the remark this answers is not in the document]
Comment (SanderVocke) [reply docs/session_format_v1.md]: Addressed in 9524229: the session-format documentation now specifies two ignored dry audio inputs, two wet outputs, and one dry MIDI input.
